airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -507,7 +507,7 @@ def create_http_requester(self, model: HttpRequesterModel, config: Config, **kwa
         )
 
         return HttpRequester(
-            name=model.name,
+            name=model.parameters.get("name", "") if model.parameters else "",
```

Comment:
I don't think that we should be making an assumption that `$parameters` will be populated with fields like `name` or `primary_key`. The main reason we see so many connector manifests where `$parameters` contained these fields was because they had to be duplicated across all multiple components. So we could have reusable interpolation pointing to one place. Ideally we should actually be deriving the `name` from the DeclarativeStream which we would want to be the single place its defined. 

We can do this by using the `kwargs` parameter while parsing the DeclarativeStream model to take `model.name` and pass it down to SimpleRetriever and HttpRequesters like such:
`retriever = self._create_component_from_model(model=model.retriever, config=config, **{"name": model.name, "primary_key": model.primary_key})`. And the child components can get them from the kwargs or set "". So that way when we are constructing components, we are actually truly referencing the stream's field and not a theoretically optional component like `$parameters`

Once we stop this field duplication that  your PR is addressing, we should be able to start constructing manifests like this:
```
New Way:
projects_stream:
  name: "projects" # we will pass this value to the requester
  primary_key: "id"
  retriever:
    $ref: "#/definitions/retriever"
    requester:
      type: HttpRequester
      url_base: "{{ config['base_url'] or 'https://app.posthog.com'}}/api/"
      http_method: "GET"
      authenticator:
        type: NoAuth
  $parameters:
    path: "projects"
  
Old Way:
projects_stream:
  retriever:
    $ref: "#/definitions/retriever"
    requester:
      type: HttpRequester
      name: "{{ parameters['name'] }}" # we reference $parameters to avoid duplicate definitions
      url_base: "{{ config['base_url'] or 'https://app.posthog.com'}}/api/"
      http_method: "GET"
      authenticator:
        type: NoAuth
  $parameters:
    name: "cohorts"
    primary_key: "id"
    path: "cohorts"
```

Hopefully that makes sense, but happy to discuss this in more detail as well

Reply:
Makes sense @brianjlai! I'd temporarily mentally rewritten how param propagation worked and assumed that the top-level parent keys would be passed down there. Fixing it up now.